src/h5.cpp

```diff
@@ -436,6 +436,9 @@ append_ret h5d_chunked_1d<T>::append_data_1d(T const *data, hsize_t nlen) {
 }
 
 template <typename T> AppendResult h5d_chunked_1d<T>::flush_buf() {
+  if (buf_n == 0) {
+    return AppendResult::OK;
+  }
```

Comment:
I'd rather avoid having more exit points from the function by wrapping the existing code in `if (buf_n > 0)` instead.

Reply:
The LLVM coding style favors early exits.

If a function has multiple exit points in nested logic, then I'd also vote to reduce the exit points.

But early-exit on pre-condition test is IMHO fine.

Reply:
"early exit" means don't execute any other code before exiting.
I'm proposing:
```cpp
template <typename T> AppendResult h5d_chunked_2d<T>::flush_buf() {
  if (buf_n > 0) {
    auto wr = DataSet.append_data_2d(reinterpret_cast<T *>(buf.data()),
                                     buf_n / sizeof(T));
    if (wr.status != AppendResult::OK) {
      return wr.status;
    }
    buf_n = 0;
  }
  return AppendResult::OK;
}
```

Reply:
> "early exit" means don't execute any other code before exiting.

I don't understand this comment. Isn't that exactly the case in my proposed version?

```cpp
template <typename T> AppendResult h5d_chunked_1d<T>::flush_buf() {
  if (buf_n == 0) {
    return AppendResult::OK;
  }
```

Nothing gets done before that return.

Reply:
For reference:

[LLVM style guide on early exit](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)

Reply:
LLVM does say "if the body of the 'if' is large". But it isn't worth further disagreement so let's get this merged.
I didn't mean to imply that your code didn't exit early, just that my suggestion exits just as early and avoids multiple exit points, which is why I prefer it.